pkg/kubectl/cmd/util/factory.go

```diff
@@ -157,6 +158,7 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 
 	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
 	flags.SetNormalizeFunc(util.WarnWordSepNormalizeFunc) // Warn for "_" flags
+	flags.Bool("disable-dynamic-discovery", false, "If true, disable calls to dynamically discover dynamic APIs")
```

Comment:
Ok, I switched to using a method parameter instead of a flag, and moved the flag out to the various commands.